lib/util.js

```diff
@@ -95,50 +98,59 @@ const meta = [
   '', '', '', '', '', '', '', '\\\\'
 ];
 
+const escapeFn = (str) => meta[str.charCodeAt(0)];
+
 // Escape control characters, single quotes and the backslash.
 // This is similar to JSON stringify escaping.
 function strEscape(str) {
-  // Some magic numbers that worked out fine while benchmarking with v8 6.0
-  if (str.length < 5000 && textRegExp.test(str))
+  if (str.length < 5000 && !strEscapeSequencesRegExp.test(str))
     return `'${str}'`;
-  if (str.length > 50) {
-    return `'${str.replace(escapeSequencesRegExp, (str) => {
-      return meta[str.charCodeAt(0)];
-    })}'`;
-  }
+  if (str.length > 100)
+    return `'${str.replace(strEscapeSequencesReplacer, escapeFn)}'`;
   var result = '';
   var last = 0;
   for (var i = 0; i < str.length; i++) {
     const point = str.charCodeAt(i);
     if (point === 39 || point === 92 || point < 32) {
-      result += `${str.slice(last, i)}${meta[point]}`;
+      if (last === i) {
+        result += meta[point];
+      } else {
+        result += `${str.slice(last, i)}${meta[point]}`;
+      }
       last = i + 1;
     }
   }
   if (last === 0) {
-    result = str;
-  } else {
+    result += str;
```

Comment:
Why the change to concatenate with an empty string? I think the assignment is a better choice.

Reply:
There was no reason for it - I was playing around. I changed it back